### PR TITLE
[SFEQS-1082] Add FCI endpoints

### PR DESCRIPTION
**Pull request by hevelius**
## Short description
This PR adds FCI endpoints and mocked payload

## List of changes proposed in this pull request
* updated package.json adding ulid (in FCI `ulid` is used as format)
* added routers and an initial test file
* added payload to mock responses
* added some PDF file with several signatureField format
* added messages CTA with front-matter

### CODE REVIEW

File: assets/fci/io-sign.yml [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+          $ref: "#/definitions/DocumentSignature"
+      qtspClauses:
+        $ref: "#/definitions/QtspClausesMetadata"
+    required:
+      - id
+      - signatureRequestId
+      - email
+      - publicKey
+      - spidSamlAssertion
+      - documentSignatures
```
Comment (gunzip) [inline assets/fci/io-sign.yml:398]: we don't need these

File: assets/fci/io-sign.yml [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+    properties:
+      id:
+        $ref: "#/definitions/SignatureId"
+      signatureRequestId:
+        $ref: "#/definitions/SignatureRequestId"
+      email:
+        $ref: "#/definitions/Email"
+      publicKey:
+        $ref: "#/definitions/PublicKey"
+      spidSamlAssertion:
```
Comment (gunzip) [inline assets/fci/io-sign.yml:384]: we don't need these here

File: assets/fci/io-sign.yml [hunk 3]
```diff
@@ -0,0 +1,385 @@
+info:
```
Comment (gunzip) [inline assets/fci/io-sign.yml:1]: you need to add a security section: https://github.com/pagopa/io-backend/blob/master/api_backend.yaml#L10

File: assets/fci/io-sign.yml [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+        - required: true
+          name: body
+          in: body
+          schema:
+            $ref: "#/definitions/CreateSignatureBody"
+      consumes:
+        - application/json
+      produces:
+        - application/json
+swagger: "2.0"
```
Comment (gunzip) [inline assets/fci/io-sign.yml:120]: may you move this line at the beginning?

File: config/config.json [hunk 5]
```diff
@@ -3,5 +3,8 @@
     "attrs": {
       "name": "Gian Maria"
     }
+  },
+  "messages": {
+    "withCTA": true
```
Comment (shadowsheep1) [inline config/config.json:8]: Did you test if adding this kind of messages it won't break the E2E test?

File: package.json [hunk 6]
```diff
@@ -24,7 +24,7 @@
   "api_eu_covid_cert": "https://raw.githubusercontent.com/pagopa/io-backend/v9.13.1-RELEASE/api_eucovidcert.yaml",
   "api_sicilia_vola": "assets/SiciliaVola.yml",
   "api_cdc": "https://raw.githubusercontent.com/pagopa/io-app/master/assets/CdcSwagger.yml",
-  "api_fci": "https://raw.githubusercontent.com/pagopa/io-backend/v9.6.0/api_io_sign.yaml",
+  "api_fci": "https://raw.githubusercontent.com/pagopa/io-backend/60c472552814b6815ea146ca18ddfe579f671452/api_io_sign.yaml",
```
Comment (shadowsheep1) [inline package.json:27]: Aren't fci specs now in production? Do we have a tag to use or we must still point to a specific branch or commit?

File: src/payloads/features/fci/signature-request.ts [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+export const EXPIRED_SIGNATURE_REQUEST_ID = ulid() as NonEmptyString;
+export const WAIT_QTSP_SIGNATURE_REQUEST_ID = ulid() as NonEmptyString;
+export const SIGNED_SIGNATURE_REQUEST_ID = ulid() as NonEmptyString;
+export const SIGNED_EXPIRED_SIGNATURE_REQUEST_ID = ulid() as NonEmptyString;
+export const DOSSIER_ID = ulid() as NonEmptyString;
+export const SIGNATURE_ID = ulid() as NonEmptyString;
+
+const now = new Date();
+const mockQrCodeUrl = "https://gist.githubusercontent.com/lucacavallaro/a3b9d5305cc6e2c9bdfb6ec1dc28fd96/raw/26799f357ff712396cdbc4f862a13099758e89d3/qr-code.png" as NonEmptyString;
+const basePath = "http://127.0.0.1:3000";
```
Comment (shadowsheep1) [inline src/payloads/features/fci/signature-request.ts:23]: Why don't you use `serverUrl` from [here](https://github.com/pagopa/io-dev-api-server/blob/ddf29ed78a3707ad0498b12e8d806c4108d736b2/src/utils/server.ts#L43)

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: config/config.json (hunk 5 above)
Comment (hevelius) [reply config/config.json:8]: Which 2e2 tests? `withCTA` is not a kind of message but a config value used to populate the message list

Comment (shadowsheep1) [reply config/config.json:8]: Well we use this [custom config](https://github.com/pagopa/io-app/blob/c5a0af370a5629584e6cdc2bae533326466c04cf/scripts/api-config.json#L18) to run E2E tests, and it's set to `true` so this won't impact. Anyway just to ask, you are adding, at the end of the list, FCI messages. Did you double check not to brake E2E tests?

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: src/populate-persistence.ts [hunk 8]
```diff
@@ -172,6 +179,42 @@ const createMessages = (
...
+      )
+    );
+    output.push(
+      getNewMessage(
+        customConfig,
+        `Comune di Controguerra - Carta d'identità Elettronica - Documento Firmato [90 days expired]`,
+        frontMatterCTAFCISignatureRequestSignedExpired +
+          messageFciSignedMarkdown
+      )
+    );
```
Comment (shadowsheep1) [inline src/populate-persistence.ts:217]: Should we add a guard with a custom config as we did for other kinds of messages?

For example:

```
 customConfig.messages.medicalCount > 0 &&
    range(1, customConfig.messages.medicalCount).forEach(count => {
      const baseMessage = medicalMessage(count);
      const attachments: ReadonlyArray<MessageAttachment> = [
        {
          name: "prescription A",
          content: "up, down, strange, charm, bottom, top",
          mime_type: "text/plain"
        },
        {
          name: "prescription B",
          content: barcodeReceipt,
          mime_type: "image/svg+xml"
        }
      ];
      output.push({
        ...baseMessage,
        content: {
          ...baseMessage.content,
          subject: `💊 medical prescription with attachments - ${count}` as MessageSubject,
          attachments
        }
      });
    });
```

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**